### PR TITLE
Chore: [M3-6828]: Cloud Manager Linting POC

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This POC implements the new linting for the package/manager codebase. Eventually we can extend the config to other packages but this is a start. In essence it does nothing but warn about linting improvements and commits those improvements to a few files as an example. It could be merged as is if everyone is comfortable with it, and engineers will be responsible to `fix` the linting suggestions in the files they're working in (one shortcut away!), or ignore them until we lint the codebase as a whole, or at least portions of it. 

Let's recap the [benefits](https://eslint-plugin-perfectionist.azat.io/guide/why):

- Readability: Finding declarations in a sorted, large list is a little faster. Remember that you read the code much more often than you write it.
- Maintainability: Sorting imports and properties is considered a good practice in software development, contributing to code quality and consistency across the codebase.
- Code Review and Collaboration: If you set rules that say you can only do things one way, then no one will have to spend time thinking about how to do it.
- Code Uniformity: When all code looks exactly the same, it is very hard to see who wrote it, which makes achieving the lofty goal of collective code ownership easier.
- Aesthetics: This not only provides functional benefits, but also gives the code an aesthetic appeal, visually pleasing and harmonious structure. Take your code to the beauty salon!

## Structure
The linting alphabetizes everything, from type & interfaces (sorry @bnussman it's a pure `asc` sorting for those, no grouping of functions but also it makes sense to keep it that way), to imports & exports, arguments and arrays element. The plugin is extremely smart at not sorting such instances of spread properties and keeps comments very much aligned with their relevant code. 

The most customization is done to the grouping of imports which is a very prominent section of the file structure, outlined in the screenshot before for a file with many imports). I do like the spacing between groups as it helps locating import sections easily. It's all customizable, to a certain extent, and I spent a fair amount of time thinking about those and configuring groups, but it's not perfect, nor will it ever be. It's also quite subjective.

Screenshot 2023-06-30 at 4 35 55 PM

## Testing
I recommend pulling this code, update your dependencies and open a variety of files to get a holistic sense of the experience developing, updating and writing code while getting those suggestions. 

## Next steps
We have a fe options as mentioned above
- Merge this as it and update as we go
- Lint the whole thing and make it an error instead of warning (I'd like to keep it a warning but it also means we have to be diligent either linting on our PRs or respecting the rules while developing
- Lint only portions of the codebase at a time to be careful not to introduce potential bugs the plugin could introduce

### CODE REVIEW

File: packages/manager/.eslintrc.js [hunk 1]
```diff
@@ -37,92 +108,126 @@ module.exports = {
...
+          'unknown',
+        ],
+        'newlines-between': 'always',
+      },
+    ],
+    'perfectionist/sort-interfaces': 'warn',
+    'perfectionist/sort-jsx-props': 'warn',
+    'perfectionist/sort-map-elements': 'warn',
+    'perfectionist/sort-named-exports': 'warn',
+    'perfectionist/sort-named-imports': 'off',
```
Comment (jaalah-akamai) [inline packages/manager/.eslintrc.js:211]: Just making a note to maybe set this to `warn` as we talked about earlier

File: packages/manager/.eslintrc.js [hunk 2]
```diff
@@ -37,92 +108,126 @@ module.exports = {
...
+      'warn',
+      {
+        'custom-groups': {
+          type: {
+            react: ['react', 'react-*'],
+            src: ['src*'],
+          },
+          value: {
+            react: ['react', 'react-*'],
+            src: ['@mui/**/*', 'src/**/*'],
```
Comment (jaalah-akamai) [inline packages/manager/.eslintrc.js:185]: This is up for debate with team: I think @mui should be grouped with the third-parties but I understand you're point in wanting to keep the components together and that with files with less imports, the spacing my be a little awkward.

**Review by jaalah-akamai — APPROVED** [review]
Overall I like this! Here are my 3 proposals:
1. Leave warnings in place and update as we make changes: This _may_ cause overly complicated diffs for simple changes, and make it hard to understand what the real significant changes are.
2. Do a sweeping commit in one PR: May introduce issues as well as force open PRs to fix conflicts.
3. Do your normal updates and for the files that changed, create a separate PR just for linting: Definitely extra work but still incremental as opposed to a sweeping change. PRs are cheap

File: packages/manager/.eslintrc.js (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/.eslintrc.js:211]: So i turned it off according to their recommendations: https://eslint-plugin-perfectionist.azat.io/rules/sort-named-imports#%F0%9F%93%96-rule-details

i think the sort imports supersedes; not sure what drawback leaving it on

File: packages/manager/.eslintrc.js (hunk 2 above)
Comment (abailly-akamai) [reply packages/manager/.eslintrc.js:185]: Fair enough, happy to reconsider it based on more feedback!

File: packages/manager/.eslintrc.js (hunk 1 above)
Comment (jaalah-akamai) [reply packages/manager/.eslintrc.js:211]: Ok that's interesting they suggest that, it appeared to work fine turned on - but I'm fine manually sorting this as I've always done 👍

Comment (abailly-akamai) [reply packages/manager/.eslintrc.js:211]: If you noticed an improvement then we should maybe try it? Would be nice if the maintainer specified what kind of conflict we could encounter..

File: packages/manager/.eslintrc.js (hunk 2 above)
Comment (bnussman-akamai) [reply packages/manager/.eslintrc.js:185]: I agree that @mui should be grouped with the third-parties

**Review by hkhalil-akamai — APPROVED** [review]
After trying these changes on one of the PRs I'm working on, I'm personally a supporter of this concept. It's like the old saying "[`gofmt`'s style is no one's favorite, yet `gofmt` is everyone's favorite.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=8m43s)" -- having consistent code style is definitely a step forward in promoting code quality.

My vote is for a sweeping change PR as suggested by @jaalah-akamai -- better to get rid of those masses of squiggly lines then let them lurk around for probably forever.

File: packages/manager/.eslintrc.js (hunk 1 above)
Comment (abailly-akamai) [reply packages/manager/.eslintrc.js]: I added the rule back. It does not seem to be an issue at all 👍

File: packages/manager/.eslintrc.js (hunk 2 above)
Comment (abailly-akamai) [reply packages/manager/.eslintrc.js]: Done, it's back with third parties

**Comment by abailly-akamai** [comment]
Thanks for the feedback so far! FWIW, I have not had any performance issue with my VScode during linting but that's a valid concern. My take is that if it becomes a problem we can always turn everything back off in the future and go back to business as usual. 

A couple outstanding questions remain, which I'd like to make decisions on, for which I'd like @jaalah-akamai to be the mighty ruler:
1. **Types import**: they sit at the top (which is the default config) and I find it to be good for me, but @bnussman-akamai has objections.
2. **Deploy**: sweeping change VS incremental changes. Both have pros & cons. I like the sweeping change as well, but only we have the confidence in our regression tooling, because the diff will be so big it will be impossible to spot a possible issue. After spending time with the plugin it looks like it's smart enough to do everything right, but who knows what could happen that wouldn't be spotted by either unit or e2e suites. All in all i want to do what's the least distracting to the team. Bonus point for a sweep: @bnussman-akamai is going away for a week so there will be twice less PRs with merge conflict 🤣

**Review by mjac0bs — APPROVED** [review]
I'm later to the game here, but voicing my support of pretty, alphabetized and organized code. (It makes the library volunteer in me feel like I'm looking at books! 📚 😄 )

- I like the line breaks in imports for readability. 
- I am used to seeing React imports at the top, so types throws me off a little, but not enough to care that much. 
- I support one sweeping PR change as long as we merge that change into `develop` with enough time for it to sit a while and allow us to time to catch any issues that may arise. (e.g. let's not merge it right before cutting a release).

**Review by cpathipa — APPROVED** [review]
Nice work! I'm in favor of this approach, as it improves readability, especially when dealing with larger files and imports.

Here is my feedback:

- IMO we don't need a line break between React packages and third party packages, I feel like we could consolidate them to one category.
- We could revisit the order of imports something like
      - React / Third party imports
      - State (shared state hooks / react query /  mutations / API interaction hooks)
      - External components (example: src/**)
      - Internanal components (example:  ./file )
      - Utils

**Comment by cpathipa** [comment]
Can we hold on merging until [this PR](https://github.com/linode/manager/pull/9341) merges ? Thinking we could avoid merge conflicts.

**Review by hana-linode — APPROVED** [review]
I agree with Chandra on merging the react and third party package imports

I'm leaning more toward a sweeping change since no code is actually changing (or shouldn't be). And if an import is missing or wrong, I think we would find out pretty quickly.

**Review by jdamore-linode — APPROVED** [review]
I like these changes @abailly-akamai ! I don't see any reason why they shouldn't apply to the Cypress code as well 👍

**Pull request merged**